File: OrderPaperApp/wwwroot/app/models/Items.ts
export class CpdBillItem {
    business_item_id: number;
    business_item_title: string;
    short_title: string;
    bill_number: string;
    member_original_name: string;
}

export class CpdMotionItem {
    business_item_id: number;
    business_item_title: string;
    title: string;
    motion: string;
    member: string;
    date: string;
}

export class CpdReportItem {
    business_item_id: number;
    business_item_title: string;
    title: string;
    shoulder: string;
    committee: string;
    latestEvent: string;
    government_response: string;
}

export class Item {
    BusinessItemId: number;
    Sequence: number;
    Type: string;
    Title: string;
    IsGrouppedItem: boolean;
    HasLine: boolean;
    IsNew: boolean;
    IsBusinessItem: boolean;
    Details: string;
    CpdTitle: string;
    constructor() {
        this.IsNew = true;
        this.Title = "Untitled";
    }
}

export class SubHeadingItem extends Item {
    FullLine: string;
    Col1a: string;
    Col2a: string;
    Col3a: string;
    Col4a: string;
    Col1b: string;
    Col2b: string;
    Col3b: string;
    Col4b: string;
    constructor() {
        super();
        this.Type = "Subheading";
        this.Title = "Sub heading";
    }
}

export class MotionItem extends Item {
    Date: string;
    Member: string;
    Speeches: string;
    Motion: string;
    CpdDate: string;
    CpdMotion: string;
    CpdMember: string;
    PrintSequence: number;
    constructor() {
        super();
        this.Type = "Motion";
        this.IsBusinessItem = true;
    }
}

export class BillItem extends Item {
    Number: string;
    Member: string;
    Stage: string;
    IsCurrentSittingWeek: boolean;
    IsFollowingSittingWeek: boolean;
    IsMajorityAmendments: boolean;
    IsConsiderationItem: boolean;
    IsBlank: boolean;
    Indication: string;
    Speeches: string;
    LatestEvent: string;
    CpdNumber: string;
    CpdMember: string;
    DocumentId: string;
    Url: string;
    constructor() {
        super();
        this.Type = "Bill";
        this.IsBusinessItem = true;
    }
}

export class ReportItem extends Item {
    Shoulder: string;
    Committee: string;
    LatestEvent: string;
    Speeches: string;
    GovernmentResponse: string;
    CpdShoulder: string;
    CpdCommittee: string;
    CpdLatestEvent: string;
    CpdGovernmentResponse: string;
    constructor() {
        super();
        this.Type = "Report";
        this.IsBusinessItem = true;
    }
}

export class LineItem extends Item {
    constructor() {
        super();
        this.Type = "Line";
        this.IsNew = false;
    }
}

export class GroupItem extends Item {
    From: number;
    To: number;
    Items: Array<Item>;
    constructor() {
        super();
        this.Items = new Array<Item>();
        this.Type = "Group";
    }
}

File: OrderPaperApp/Models/Items.cs
using System.Collections.Generic;
using System;

namespace OC.OrderPapersWeb.Models
{
    public class Item
    {
        public int Sequence { get; set; }
        public string Type { get; set; }
        public string Title { get; set; }
    }

    public class MotionItem : Item
    {
        public DateTime Date { get; set; }
        public string Summary { get; set; }
        public string Member { get; set; }
        public string Speeches { get; set; }
        public string Motion { get; set; }
        public MotionItem() {
            this.Type = "Motion";
        }
    }

    public class BillItem : Item
    {
        public string Number { get; set; }
        public string Member { get; set; }
        public string Stage { get; set; }
        public bool IsCurrentSittingWeek { get; set; }
        public bool IsFollowingSittingWeek { get; set; }
        public bool IsMajorityAmendments { get; set; }
        public bool IsExtendedSittingHours { get; set; }
        public string Speeches { get; set; }
        public string LatestEvent { get; set; }

        public BillItem() {
            this.Type = "Bill";
        }
    }

    public class ReportItem : Item
    {
        public string Shoulder { get; set; }
        public string Committee { get; set; }
        public string LatestEvent { get; set; }
        public ReportItem() {
            this.Type = "Report";
        }
    }

    public class LineItem : Item
    {
        public LineItem() {
            this.Type = "Line";
        }
    }

    public class GroupItem : Item
    {
        public List<Item> Items { get; set; }
        public GroupItem() {
            this.Items = new List<Item>();
            this.Type = "Group";
        }
    }
}
